src/foam/nanos/auth/PreventDuplicateEmailLoginService.js

```diff
@@ -29,9 +29,15 @@ foam.CLASS({
       name: 'getUser',
       javaCode: `
         DAO userDAO = ((DAO) getX().get("localUserUserDAO")).where(
-          OR(
-            EQ(User.EMAIL, identifier.toLowerCase()),
-            EQ(User.USER_NAME, identifier)));
+          AND(
+            OR(
+              EQ(User.EMAIL, identifier.toLowerCase()),
+              EQ(User.USER_NAME, identifier)
+            ),
+            EQ(User.LIFECYCLE_STATE, LifecycleState.ACTIVE)
```

Comment:
@adamvy only username is unique.  unique email is enforced by the application by enabling the associated rules.  
username is only used for login, nothing internal depends on it. 
So after a user is 'deleted', in the base foam system, I don't see any reason no to allow reuse.  If an application wants another strategy then it can impose that.  
The only compromise here would be have this check be a rule like duplicate email, and then it's easy for an application to enable/disable.

Reply:
This doesn't check if a user is deleted though, it checks if they are ACTIVE.  Meaning we could have a PENDING user and an ACTIVE user with the same email.  But there isn't a unique email check on moving a user from PENDING to ACTIVE

Reply:
Or in theory even many PENDING users all with the same email. Maybe that's desired? Just feels surprising I guess.

Reply:
Condition removed.  This DAO is used in nspec uniqueUserService which itself adds the condition LifecycleState != DELETED. And since this condition is in the nspec it makes it easy for an application to override.